### PR TITLE
Be: Entities, services and GET, POST, PUT and DELETE endpoints for quizes and users

### PR DESCRIPTION
**Pull request by dethurah**
Sets up the required entities and their relations with Hibernate

### CODE REVIEW

**Comment by dethurah** [comment]
Not all endpoints are implemented yet. Some are still missing.
The endpoints seem to work, but there might still be a risk of some bugs. Tests are needed.

File: src/main/java/Quiz/dto/QuestionIdDto.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.util.Collection;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class QuestionIdDto {
+    private int id;
+    private String description;
+    private Collection<QuestionOptionIdDto> options;
+
+    public int getId() {
```
Comment (siffravn) [inline src/main/java/Quiz/dto/QuestionIdDto.java:13]: Please use project lombok so we can be free of basic getters and setters in Moldes and DTOs

        
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>1.18.10</version>
        </dependency>

File: src/main/java/Quiz/QuizService.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        Quiz quiz = session.get(Quiz.class, id);
+
+        if (quiz == null)
+            return Response.status(Response.Status.NOT_FOUND).build();
+
+        QuizIdDto dto = new ObjectMapper().convertValue(quiz, new TypeReference<QuizIdDto>(){});
+        session.close();
+        return Response.status(Response.Status.OK).entity(dto).build();
+    }
+
```
Comment (siffravn) [inline src/main/java/Quiz/QuizService.java:35]: Can we create QuizDAO and extract all hibernate interaction from the service to this? In this way the DAO is concerned with Hibernate and the service is concerned with REST :) 
Same applies to User

File: src/main/java/Util/HibernateUtil.java [hunk 3]
```diff
@@ -1,3 +1,5 @@
+package Util;
+
```
Comment (siffravn) [inline src/main/java/Util/HibernateUtil.java:2]: Very nice util class for hibernate!

**Review by siffravn — CHANGES_REQUESTED** [review]

File: src/main/java/Quiz/QuizService.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            quizIdSaved = (int) session.save(quiz);
+            tx.commit();
+        } catch (HibernateException ex) {
+            if (tx != null)
+                tx.rollback();
+            ex.printStackTrace();
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        } finally {
+            session.close();
+        }
```
Comment (siffravn) [inline src/main/java/Quiz/QuizService.java:60]: Can we create QuizDAO and extract all hibernate interaction from the service to this? In this way the DAO is concerned with Hibernate and the service is concerned with REST :)
Same applies to User

File: src/main/java/Quiz/dto/QuestionIdDto.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        this.description = description;
+    }
+
+    public Collection<QuestionOptionIdDto> getOptions() {
+        return options;
+    }
+
+    public void setOptions(Collection<QuestionOptionIdDto> options) {
+        this.options = options;
+    }
```
Comment (siffravn) [inline src/main/java/Quiz/dto/QuestionIdDto.java:35]: Please use project lombok so we can be free of basic getters and setters in Moldes and DTOs

    
    <dependency>
        <groupId>org.projectlombok</groupId>
        <artifactId>lombok</artifactId>
        <version>1.18.10</version>
    </dependency>

**Review by siffravn — CHANGES_REQUESTED** [review]

File: src/main/java/Util/HibernateUtil.java (hunk 3 above)
Comment (siffravn) [inline src/main/java/Util/HibernateUtil.java:2]: Very nice util class for hibernate :)

File: src/main/java/Quiz/QuizService.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("quiz")
+public class QuizService {
```
Comment (siffravn) [inline src/main/java/Quiz/QuizService.java:19]: Please add unit test to services

**Review by siffravn — CHANGES_REQUESTED** [review]

**Comment by siffravn** [comment]
Nice work! The PR is a bit big, could have been nice if it was scoped by domain 😃

File: src/main/java/Quiz/QuizService.java (hunk 4 above)
Comment (dethurah) [reply src/main/java/Quiz/QuizService.java]: I have now added a QuizDao and UserDao (and interfaces for each), and tried to refactor hibernate logic into those :)

File: src/main/java/Quiz/dto/QuestionIdDto.java (hunk 5 above)
Comment (dethurah) [reply src/main/java/Quiz/dto/QuestionIdDto.java]: I have now added Lombok and removed all getters and setters. For now, I have only used the @Data annotation from Lombok, but perhaps we should use more detalied annotaions such as @Getter and @Setter.

File: src/main/java/Util/HibernateUtil.java (hunk 3 above)
Comment (dethurah) [reply src/main/java/Util/HibernateUtil.java:2]: Thanks :)

**Review by siffravn — APPROVED** [review]

**Pull request merged**